### PR TITLE
WebXR initialization

### PR DESCRIPTION
**Pull request by asajeffrey**
Initialize a WebXR device registry in Servo.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because this is hidden plumbing at the moment

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23675)

### CODE REVIEW

File: components/servo/Cargo.toml [hunk 1]
```diff
@@ -78,6 +78,7 @@ webrender_api = {git = "https://github.com/servo/webrender", features = ["ipc"]}
 webdriver_server = {path = "../webdriver_server", optional = true}
 webvr = {path = "../webvr"}
 webvr_traits = {path = "../webvr_traits"}
+webxr-api = "0.0.1"
```
Comment (Manishearth) [inline components/servo/Cargo.toml]: Can we use git deps for now? I don't think we're at a stage yet where we should be using versioned crates.

**Review by Manishearth — APPROVED** [review]

File: components/servo/Cargo.toml (hunk 1 above)
Comment (asajeffrey) [reply components/servo/Cargo.toml]: Ok

**Comment by asajeffrey** [comment]
Oh rats, there's merge conflicts.

**Comment by asajeffrey** [comment]
Oops, pushed changes to this branch by mistake. @bors-servo r=Manishearth

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Comment by asajeffrey** [comment]
Rebased. @bors-servo r=Manishearth

**Comment by jdm** [comment]
@bors-servo retry
#23693

**Pull request merged**